### ISSUE TITLE
Add rich text functionality to CardDiscussionInput

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#564]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue is to investigate various Vue/JS/TS plugins to allow us to add markdown and rich text functionality to [CardDiscussionInput.vue](https://github.com/activist-org/activist/blob/main/frontend/components/card/discussion/CardDiscussionInput.vue). We should consider a few possible options, and then this issue can shift to implementing the changes 😊

### Contribution

Would be happy to discuss options with people and work on implementation or support someone with it!

**Comment by andrewtavis** [comment]
Looking for options for this, for RTE we have the following:

- [tiptap](https://github.com/ueberdosis/tiptap)
- [tinymce](https://github.com/tinymce/tinymce)

For markdown we could use the following:

- [vue-markdown-editor](https://github.com/code-farmer-i/vue-markdown-editor)
- [Mavon Editor](https://github.com/hinesboy/mavonEditor)

**Comment by andrewtavis** [comment]
A quick note on this, it'd be great if we'd also do some styling as a part of this such that the options are only present for a text field if it's currently active in the same what that it is for GitHub :)

**Comment by andrewtavis** [comment]
Note that this issue is now only for RTE, and likely the solution we'll be using [tiptap](https://github.com/ueberdosis/tiptap) :)

**Comment by bharath637462** [comment]
@andrewtavis can i work on this?

**Comment by Mister-Tofu-tech** [comment]
@andrewtavis . Hopefully I can help out with this.

**Comment by andrewtavis** [comment]
Give it a shot, @Mister-Tofu-tech :) Happy to support! We can also do a quick call to discuss if that would help.

**Comment by andrewtavis** [comment]
Hey @Mister-Tofu-tech 👋 Let me know what the status here is :) We're getting to a point where this issue will be a bit higher priority. I can also take it over if you're pressed for time :)

**Comment by Mister-Tofu-tech** [comment]
Hey @andrewtavis , sorry I wasn't able to make progress towards this issue. Life has been busy.
Hopefully, I can work on something with lower priority?

**Comment by andrewtavis** [comment]
Yes, no stress @Mister-Tofu-tech! Feel free to check out the good first issues or other with help wanted :)

**Comment by andrewtavis** [comment]
Potentially looking into one of the social share issues would work on your end? We're looking for people to expand the page share component. Would be great to get some support there!

**Comment by Mister-Tofu-tech** [comment]
@andrewtavis I think I can take this one. Feel free to assign it to me.

**Comment by andrewtavis** [comment]
Sure thing, @Mister-Tofu-tech :)

**Comment by andrewtavis** [comment]
Let me know if you want to discuss!

**Comment by UnknownSean8** [comment]
Hi @andrewtavis, I am back, I am happy to work on this if this is still available. Thanks!

**Comment by andrewtavis** [comment]
Would be awesome to get your support here, @UnknownSean8! Assigning you :) Happy to discuss in the sync and pair on it if need be 😊

**Comment by UnknownSean8** [comment]
Is there anywhere I can find the component on the page? I've tried the discussion page on both the events and organizations page but there is none. Thanks!

**Comment by andrewtavis** [comment]
Ya it's not on those pages at the moment. You'd be welcome to put it on one of those pages for the time being, @UnknownSean8! I can remove it during review :)

**Comment by UnknownSean8** [comment]
![Image](https://github.com/user-attachments/assets/59dc8788-d3de-4758-85f5-e15d89a0cf17)

So far it looks pretty good, just without preview for now, thew editor itself is the preview.

**Comment by andrewtavis** [comment]
Amazing, @UnknownSean8! Feel free to open a PR! Looking forward to checking this out 😊

**Comment by andrewtavis** [comment]
Closing this issue now that #1135 has been merged. All further work for the discussion input will be handled by #1145 🚀

Thanks @UnknownSean8!
